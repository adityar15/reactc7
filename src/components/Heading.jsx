import React from 'react'

// functional components
export default function Heading({title}) {
  return (
    <h1 className="text-lg text-red-600">{title}</h1>
  )
}


