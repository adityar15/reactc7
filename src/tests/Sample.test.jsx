import React from 'react'
import renderer from 'react-test-renderer'
import Heading from '../components/Heading'

import { expect, test } from 'vitest'

// tools used
// react-test-renderer happy-dom jsdom 

// function toJson(component) {
//   const result = component.toJSON()
//   expect(result).toBeDefined()
//   expect(result).not.toBeInstanceOf(Array)
//   return result
// }
test('Checks for heading component and makes sure that it is displayed correctly', () => {

  const component = renderer.create(<Heading title="Adi" />);
  let tree = component.toJSON()
  console.log("to json", tree)
  expect(tree).toMatchSnapshot();


  // If you need to test different states or props, you can add more test cases here.
});

