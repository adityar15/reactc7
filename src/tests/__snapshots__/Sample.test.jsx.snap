// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading component 1`] = `
<h1
  className="text-lg text-red-600"
>
  Adi
</h1>
`;
